src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -542,18 +542,20 @@ static boost::optional<detail::STVar> parseLeaf (
 
                         std::string const strValue (account.asString ());
 
-                        if (value.size () == 40) // 160-bit hex account value
-                            uAccount.SetHex (strValue);
+                        // If we have what looks like a 160-bit hex account
+                        // value, try it and see if it's valid.
+                        bool const success = (value.size () == 40)
+                            ? uAccount.SetHex (strValue)
+                            : false;
 
+                        if (!success)
```

Comment:
That assumes your reading speed and cognitive space are infinite.  :-D  But sure, this is a micro detail!

(But do fix up the `? : false`, that's just silly.  :-P)